### ISSUE TITLE
Multiple customs posts type and function add_local_field_groups

### ISSUE THREAD

**Issue by psntr** [Hube2/acf-post2post#22]
Hi,

In the past I have used the plugin with Object Post. This time I had to do it in another way. 

I have 3 customs posts types: 

- Exhibition
- Work
- Artist

Those 3 posts types have sometimes 2 or 3 relationship field which link them to either Exhibition, Work or Artist custom post type content. 

First of all I've created the field with the ACF UI, I made sure to use same "name" for the relationship field. For example:

![exhibition_acf](https://user-images.githubusercontent.com/20617539/49085014-52c95880-f251-11e8-83d4-63159d8a8ff7.png)
![work_acf](https://user-images.githubusercontent.com/20617539/49085015-52c95880-f251-11e8-9f17-e8b8c76a2c32.png)
![artist_acf](https://user-images.githubusercontent.com/20617539/49085017-5361ef00-f251-11e8-97b7-9ee30d9cc755.png)

Then in a new WordPress install, I used the function "add_local_field_groups" in my functions.php. It seems to work without problems, since the field are showing up and save the data. 

The problem is, as you might have expected, they don't sync between them. 

Do you know why? I have tried first without the plugin with the DOC, found here: https://www.advancedcustomfields.com/resources/bidirectional-relationships/

Then your recommandation found here (also tried the GitHub link provided - maybe I didn't do it correctly):
https://support.advancedcustomfields.com/forums/topic/bidirectional-relationships-not-updating-both-sides/

**Comment by Hube2** [comment]
This could have something to do with where and when you are adding the field groups in PHP.

In what file is your code?

When is it executed? Are you using a specific hook?

**Comment by psntr** [comment]
The field groups in PHP are in a file called _bones.php_ which is directly called in the beginning of my _functions.php_ 

So in the _bones.php_ I have all the functions hook.  For the ACF field group I have this:

```
function my_acf_add_local_field_groups() {
if( function_exists('acf_add_local_field_group') ):
…
endif;
}
```

After that, in my  _functions.php_  I called the file containing my hooks:

`require_once( 'library/bones.php' );`

After the require_once, I have this function:

```
function rt_launch() {
require_once( 'library/custom-post-type.php' );

  // launching operation cleanup
  add_action( 'init', 'rt_head_cleanup' );
  // A better title
  add_filter( 'wp_title', 'rw_title', 10, 3 );
  // remove WP version from RSS
  add_filter( 'the_generator', 'rt_rss_version' );
  // remove pesky injected css for recent comments widget
  add_filter( 'wp_head', 'rt_remove_wp_widget_recent_comments_style', 1 );
  // clean up comment styles in the head
  add_action( 'wp_head', 'rt_remove_recent_comments_style', 1 );
  // clean up gallery output in wp
  add_filter( 'gallery_style', 'rt_gallery_style' );
  
  // Init ACF
 add_action('acf/init', 'my_acf_add_local_field_groups');
}
add_action( 'after_setup_theme', 'rt_launch' );
```
Below that group, there is other functions as such as:
`add_image_size` 
`update_option` (size of some thumbnails)
`$wp_customize->remove_section`

—
Basically it's called after_setup_theme, as the hook is in my rt_launch function.  What do you think?

**Comment by Hube2** [comment]
this is how it appears to me

On the Exhibition post type you have a relationship field name "artist_related", but on the Artist page you ave a field named "exhibition_related". If these two fields need to be bidirectional with each other then they are not named the same.

Exhibition and Work share a field named "artist_related", these two fields will be bidirectional. I don't think this is what you had in mind.

This means that selecting an "Artist" when updating a "Exhibition" the artist will not be updated because it does not share a commonly name field on the artist post type.

When I create fields to use with this plugin I try to name them using the post types that will be related.

for example, looking at your field setup you need 4 fields (7 actually counting the fields for each post type):
exhibition_to_work (used on the exhibition and work post types)
exhibition_to_artist (used on the exhibition and artist post types)
work_to_artist (used on the work and artist post types)
work_to_work (uses only on the work post type)

**Comment by psntr** [comment]
I took your comments in account and renamed the field accordingly. And now it works both ways! 
You were right, I was a bit confused on how to use the field name for the relationship, but it's quite clear now. 

Thank you for your help!
